httputil/useragent.go

```diff
@@ -60,7 +60,10 @@ func SetUserAgentFromVersion(version string, extraProds ...string) {
 	// xxx this assumes ReleaseInfo's ID and VersionID don't have weird characters
 	// (see rfc 7231 for values of weird)
 	// assumption checks out in practice, q.v. https://github.com/zyga/os-release-zoo
-	userAgent = fmt.Sprintf("snapd/%v (%s)%s %s/%s (%s)", version, strings.Join(extras, "; "), extraProdStr, release.ReleaseInfo.ID, release.ReleaseInfo.VersionID, string(arch.UbuntuArchitecture()))
+	userAgent = fmt.Sprintf("snapd/%v (%s)%s %s/%s (%s %s)", version,
```

Comment:
@niemeyer @zyga it's supposed to be a list of `<thing>/<version-of-thing> (optional space-separated details about thing)` in decreasing order of significance, so it being e.g. `snapd/2.22 (classic) ubuntu/16.04 (amd64) linux/5.0.1 (PREEMPT POTATO)` would be fine; or the kernel could be a detail of the distribution, which would also be ok as long as we never want details about the kernel.